flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/calls/ScalarOperators.scala

```diff
@@ -984,6 +987,63 @@ object ScalarOperators {
     }
   }
 
+  def generateDot(codeGenerator: CodeGenerator,
```

Comment:
I think this method is not really necessary. We already have logic for accessing composite types (see `visitFieldAccess()`). Maybe we just have to make the methods there a bit more generic.

Reply:
In Calcite, once it sees array element access, the subsequent field access is translated into DOT RexCall, not RexFieldAccess. Therefore, we need to a custom handling for the DOT RexCall.

Reply:
But I agree I should refactor the code in visitFieldAccess() to reuse it as much as possible.